File: BTS/Java/Premiere annee/fonderie/BarreCarree.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package fonderie;

/**
 *
 * @author therbron
 */
public class BarreCarree extends Barre{
    
    private double cote;

    public BarreCarree() {
        super();
        this.cote = 3.25;
        System.out.println("Constructeur defaut BarreCarre");
    }

    public BarreCarree(double cote, double densite, double longueur, String reference) {
        super(densite, longueur, reference);
        this.cote = cote;
        System.out.println("Constructeur parametre et super parametres BarreCarre");
    }

    public BarreCarree(BarreCarree autreBarre) {
        super(autreBarre);
        this.cote = autreBarre.cote;
        System.out.println("Constructeur copie BarreCarre");
    }
    
    double calculerMasse()
    {
        return longueur*densite*(cote*cote);
    }
    
}

File: BTS/C++/polymorphismeCours/BarreCarree.cpp
#include "BarreCarree.h"
#include "Barre.h"

BarreCarree::BarreCarree(const float _densite, const float _longueur, const string _referenceBarre, const float _cote)
    : Barre(_densite, _longueur, _referenceBarre), cote(_cote)
{
cout << "Je suis le constructeur a argument de BarreCarree." << endl;
}

BarreCarree::BarreCarree()
    : Barre(), cote(2)
{
    cout << "Je suis le constructeur par defaut de BarreCarree." << endl;
}

BarreCarree::BarreCarree(const BarreCarree &barreCarree)
{
    cout << "Je suis le constructeur par copie de BarreCarree." << endl;
}

BarreCarree::~BarreCarree()
{
    cout << "BarreCarree detruite." << endl;
}

float BarreCarree::calculerSection()
{
    return cote*cote;
}

float BarreCarree::calculerMasse()
{
    return longueur*densite*calculerSection();
}
